docs/Dialects/Debug.md

```diff
@@ -0,0 +1,111 @@
+# Debug Dialect
+
+This dialect provides operations and types to interleave debug information (DI) with other parts of the IR.
+
+[TOC]
+
+
+## Rationale
+
+The main goal of the debug dialect is to provide a mechanism to track the correspondence between values, types, and hierarchy of a source language and the IR being compiled and transformed. This allows simulators, synthesizers, and other debugging tools to reconstruct a source language view into the processed hardware and allow for easier debugging by humans.
+
```

Comment:
Supreme nit: 

Consider either line wrapping to 80-column or using 1-sentence-per-line. Each has trade-offs. The former is readable, but constant line wrapping makes diffs hard. The latter makes for great diffs.

Reply:
Oh yeah excellent point. Since this is going to be rendered to a webpage, picking a fixed column is a bit weird. One sentence per line though sounds great. Probably should go change some of the other rationale documents which contain entire paragraphs on a single line.

Reply:
Okay I tried the 1-sentenced-per-line thing and it makes me very uneasy. Going to do the 80 column wrap.